bitwarden/clients
# Reduce blinking of notification bar

## Type of change

## Objective
This aims to reduce the blinking of the save/update password prompt. First by only re-opening the bar if the actual tab changed and to not re-open if the i-frame is still visible.
If the page re-renders or navigates elsewhere, we still need to redisplay it. Blinking (re-render) in this case cannot be prevented.

Asana-Ticket: https://app.asana.com/0/1169444489336079/1177106211624780

Closes #1243

Shout-out to @qwx 💪 for figuring out that the bar gets affected by other tabs 🎉 

## Code changes
- **src/background/tabs.brackground.ts**: 
  - Used proper types (chrome.tabs.XXX)
  - Pass the tab that changed to the `checkNotificationQueue` (prevents flickering when other tabs update)

- **src\content\notificationBar.ts**: 
  - Set `iframe.src` so that the check in `closeExistingAndOpenBar` for the frame's existance works

## QA required
1) Enter an unsaved login so that the save password prompt (notificationBar) appears
Open a second tab with either a site you know that updates the title (I used track.toggl.com or used the script mentioned [here](https://github.com/bitwarden/browser/issues/1243#issuecomment-911346095)
Switch back to the tab with the password prompt, it should not be affected by the other tab updating
2) The notificationBar should not disappear when logging in to a site (re-render/navigate)

## Checklist
- [X] I have checked for linting errors (`npm run lint`)
- [ ] This change requires a **documentation update** (alert the documentation team in Asana)
- [ ] This change has specific **deployment requirements** (alert DevOps in Asana)

## Other repositories affected
_Add links if this is the main PR for a change._
- [ ] jslib
